porcelain/storagedeals_test.go

```diff
@@ -139,3 +140,162 @@ func TestDealMinerLs(t *testing.T) {
+	require.Equal(trp.t, trp.dealCid, c)
+
+	deal := &storagedeal.Deal{
+		Proposal: &storagedeal.Proposal{
+			Payment: storagedeal.PaymentInfo{
+				Vouchers: trp.vouchers,
+			},
+		},
+	}
+
+	return deal, nil
+}
+
+func (trp *testRedeemPlumbing) MessagePreview(_ context.Context, fromAddr address.Address, actorAddr address.Address, method string, params ...interface{}) (types.GasUnits, error) {
+	require.Equal(trp.t, trp.fromAddr, fromAddr)
+	require.Equal(trp.t, address.PaymentBrokerAddress, actorAddr)
+	require.Equal(trp.t, "redeem", method)
+	require.Equal(trp.t, []interface{}{
+		trp.expectedVoucher.Payer,
+		&trp.expectedVoucher.Channel,
+		trp.expectedVoucher.Amount,
+		&trp.expectedVoucher.ValidAt,
+		trp.expectedVoucher.Condition,
+		[]byte(trp.expectedVoucher.Signature),
+		[]interface{}{},
+	}, params)
+	return trp.gasPrice, nil
+}
+
+func (trp *testRedeemPlumbing) MessageSendWithDefaultAddress(_ context.Context, fromAddr address.Address, actorAddr address.Address, _ types.AttoFIL, _ types.AttoFIL, _ types.GasUnits, method string, params ...interface{}) (cid.Cid, error) {
+	require.Equal(trp.t, trp.fromAddr, fromAddr)
+	require.Equal(trp.t, address.PaymentBrokerAddress, actorAddr)
+	require.Equal(trp.t, "redeem", method)
+	require.Equal(trp.t, []interface{}{
+		trp.expectedVoucher.Payer,
+		&trp.expectedVoucher.Channel,
+		trp.expectedVoucher.Amount,
+		&trp.expectedVoucher.ValidAt,
+		trp.expectedVoucher.Condition,
+		[]byte(trp.expectedVoucher.Signature),
+		[]interface{}{},
+	}, params)
+	return trp.messageCid, nil
+}
+
+func TestDealRedeem(t *testing.T) {
+	tf.UnitTest(t)
+
+	addressGetter := address.NewForTestGetter()
+	cidGetter := types.NewCidForTestGetter()
+	dealCid := cidGetter()
+	messageCid := cidGetter()
+	fromAddr := addressGetter()
+	payerAddr := addressGetter()
+	channelID := *types.NewChannelID(0)
+	tooSmallVoucher := &types.PaymentVoucher{
+		Payer:   payerAddr,
+		Channel: channelID,
+		Amount:  types.NewAttoFILFromFIL(1),
+		ValidAt: *types.NewBlockHeight(10),
+	}
+	expectedVoucher := &types.PaymentVoucher{
+		Payer:   payerAddr,
+		Channel: channelID,
+		Amount:  types.NewAttoFILFromFIL(2),
+		ValidAt: *types.NewBlockHeight(20),
+	}
+	notYetValidVoucher := &types.PaymentVoucher{
+		Payer:   payerAddr,
+		Channel: channelID,
+		Amount:  types.NewAttoFILFromFIL(3),
+		ValidAt: *types.NewBlockHeight(30),
+	}
+	vouchers := []*types.PaymentVoucher{
+		tooSmallVoucher,
+		expectedVoucher,
+		notYetValidVoucher,
+	}
+	plumbing := &testRedeemPlumbing{
+		t:               t,
+		blockHeight:     types.NewBlockHeight(25),
+		dealCid:         dealCid,
+		expectedVoucher: expectedVoucher,
+		fromAddr:        fromAddr,
+		messageCid:      messageCid,
+		vouchers:        vouchers,
+	}
+
+	resultCid, err := porcelain.DealRedeem(context.Background(), plumbing, fromAddr, dealCid, types.NewAttoFILFromFIL(0), types.NewGasUnits(0))
+	require.NoError(t, err)
+
+	assert.Equal(t, messageCid, resultCid)
```

Comment:
Running the equality assertions in the stub obscures the meaning of this test. It's correct and sufficient, but it takes some unnecessary analysis on the readers part to establish what's under test.